### ISSUE TITLE
Add arm64 as supported arch in snap

### ISSUE THREAD

**Issue by tobhe** [Ericsson/codechecker#3784]
I would love to use the snap on my m2 macbook and there does not seem to be any real blocker to get it built on arm64. Having the snap binary available would be nice.

**Comment by bruntib** [comment]
Could `pip install codechecker` work as an alternative? We have that too.

**Comment by bruntib** [comment]
CodeChecker is written in Python, so it should be platform independent in this regard. The only platform dependent part is the ld-logger, but that is not supported in OSX, but `intercept-build` is available there.
@gamesh411 Could you, please, test if CodeChecker can be installed on OSX from snap? Are there any issues? Thank you!

**Comment by gamesh411** [comment]
Snap packages cannot be installed on macOS at all - snapd depends on Linux kernel features (namespaces, cgroups, squashfs, AppArmor), and Canonical has no macOS port. Only snapcraft, the build/store CLI, runs on macOS via Homebrew. So there is nothing to test on the OSX side.

The original request most likely means running the snap inside a Linux arm64 VM on an M2, which is reasonable, but arm64 is currently the smaller problem. CodeChecker actually has a version published as an amd64 snap: version 6.21.0, revision 13, last released 2022-11-11. The publishing pipeline is not up to date, though, as every run of the codechecker-snap-package workflow in the retained history has been cancelled: runs-on: ubuntu-20.04 no longer resolves to a runner, so the job waits 24h and gets killed before any step runs. snapcraft.yaml also still targets EOL core18 with a hardcoded python3.9 path.

Unbreaking publishing would need: bump the runner label, move base to core22/core24 and drop the python3.9 hardcode, then add architectures plus an ubuntu-24.04-arm build job for arm64. Alternatively, we decide the snap is not worth maintaining and point users at pip3 install codechecker or a source build, which is what macOS users should use today anyway (see docs/install_macosx.md, noting the SIP caveat for CodeChecker log).

**Comment by tobhe** [comment]
> Snap packages cannot be installed on macOS at all - snapd depends on Linux kernel features (namespaces, cgroups, squashfs, AppArmor), and Canonical has no macOS port. Only snapcraft, the build/store CLI, runs on macOS via Homebrew. So there is nothing to test on the OSX side.

Indeed, I am using Linux not OSX. Whether there is an official Ubuntu port doesn't matter IMO, I probably just shouldn't have mentioned the device. This affects all arm64 machines on Linux of course, snaps are used outside of Ubuntu too and there are a bunch of Linux arm64 machines out there (Raspberry Pi, Nvidia DGX Spark, VMs etc.).

> CodeChecker is written in Python, so it should be platform independent in this regard

Yes, and afair building it worked fine when I reported it 4 years ago. but it needs to be explicitly built and published for arm64 in snapcraft.yaml and the store.
